### ISSUE TITLE
os.Exit kills entire program not just the interpreter.

### ISSUE THREAD

**Issue by james-lawrence** [containous/yaegi#486]
I'm unsure if this is a good thing or not.

```golang
package main

import (
	"log"
)

func main() {
	log.Fatalln("fail")
}
```

**Comment by ashneverdawn** [comment]
If using yaegi to run untrusted code in a sandbox, this is definitely very undesirable behavior. 

However, I guess the question is whether or not this should be changed at the yaegi level. 

You 'could' change the way log.Fatalln() behaves without changing yaegi.
Instead of using:
`in.Use(stdlib.Symbols)`
You would use your own modified version of the standard libraries.

Of course it would be more user friendly if yaegi provided an additional version of stdlib for use with untrusted code. For example it could be something like:
`in.Use(untrustedstdlib.Symbols)`

I'd like to know whether or not yaegi wants to officially support this use case. I'm interested in using yaegi to run untrusted code and am willing to contribute. 
If not, I could make an open source wrapper library around yaegi for the purposes of running untrusted code.

**Comment by mvertes** [comment]
Complete sandboxing seems to be out of reach, as it is always possible for a pre-compiled binary module (launched by a script snippet or not) to spawn go-routines, system calls, etc. out of control from the interpreter. Achieving full control would require a different Go runtime (not only stdlib),  probably more complex and slower, and not in the philosophy of the current one.

The only fully safe option would be to isolate the untrusted code in a separate process, using fork.

Partial sandboxing, where we can audit and control execution at interpreter level is something that we support, and want to improve (see work on EvalWithContext), your suggestion regarding restricted stdlib symbols seems to fit.

**Comment by james-lawrence** [comment]
yeah I did those things after i filed the issue it was just to bring it to the attention of the maintainers.

my personal approach would be to change stdlib implementation to be as safe as we can make it.

then provide the current form as an unsafestdlib.

@mvertes safety/trustworthiness is one thing. but the os.Exit is another. the interpreted code really shouldn't crash the host application, but I am unsure how desirable it is to override it by default.

for the record, overridding os.Exit wasn't sufficient, I had to override log.Fatal* directly because the os.Exit used by log in yaegi isn't the one is the stdlib exports (for obvious reasons)
